text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,418 @@
+can override using custom cron expression. The trigger method will take an
+optional description but abstract the requirement of an actions list using
+the job or crawler name. \
+
+```
+myGlueJob.createDailyTrigger(this, 'MyDailyTrigger');
+
+myGlueJob.createHourlyTrigger(this, 'MyHourlyTrigger');
+
+myGlueJob.createWeeklyTrigger(this, 'MyWeeklyTrigger');
+
+myGlueJob.createScheduledTrigger(this, 'MyScheduledTrigger', {
+  description: 'Scheduled run for ' + crawler.name,
+  schedule: '`cron(15 12 * * ? *)'`` //``every day at 12:15 UTC`
+});
+```
+
+#### **3. Notify  Event Trigger**
+
+This type of trigger is only supported with Glue workflow. There are two types
+of notify event triggers, batching and non-batching trigger. For batching trigger
+`BatchSize` customer has to specify but for non-batching `BatchSize` will be set
+to 1. For both trigger type `BatchWindow will be default to 900 seconds`
+
+```
+myGlueJob.createNotifyEventBatchingTrigger(this, 'MyNotifyTrigger', batchSize,
+    workFlowName: workflow.name,
+    batchSize: batchSize
+);
+
+myGlueCrawler.createNotifyEventBatchingTrigger(this, 'MyNotifyTrigger', batchSize,
+    workFlowName: workflow.name,
+    batchSize: batchSize
+);
+
+myGlueJob.createNotifyEventNonBatchingTrigger(this, 'MyNotifyTrigger',
+    workFlowName: workflow.name
+);
+
+myGlueCrawler.createNotifyEventNonBatchingTrigger(this, 'MyNotifyTrigger',
+    workFlowName: workflow.name
+);
+
+```
+
+#### **4. Conditional Trigger**
+
+Conditional trigger has predicate and action associated with it. Based on
+predicate, trigger action will be executed.
+
+```
+// Triggers on Job and Crawler status
+myGlueJob.addConditionalTrigger(
+  jobs: [
+    {jobArn: "job1-arn", status: glue.JobStatus.SUCCEEDED},
+    {jobArn: "job2-arn", status: glue.JobStatus.FAILED},
+  ], crawlers: [
+    {crawlerArn: "crawler1-arn", status: glue.CrawlerStatus.SUCCEEDED},
+    {crawlerArn: "crawler2-arn", status: glue.CrawlerStatus.TIMEOUT},
+]);
+
+```
+
+### Connection Properties
+
+A `Connection` allows Glue jobs, crawlers and development endpoints to access
+certain types of data stores.
+
+* **Secrets Management
+    **User needs to specify JDBC connection credentials in Secrets Manager and
+    provide the Secrets Manager Key name as a property to the Job connection
+    property.
+
+* **Networking - CDK determines the best fit subnet for Glue Connection
+configuration
+    **The current glue-alpha-module requires the developer to specify the
+    subnet of the Connection when it’s defined. This L2 RFC will make the
+    best choice selection for subnet by default by using the data source
+    provided during Job provisioning, traverse the source’s existing networking
+    configuration, and determine the best subnet to provide to the Glue Job
+    parameters to allow the Job to access the data source. The developer can
+    override this subnet parameter, but no longer has to provide it directly.
```

Comment:
Can you provide a little more technical detail on how you plan to do this? If you intend to make an API call to accomplish this, I would strongly suggest storing the value in `context` so that it only has to be done the first time.

Reply:
Janny will add more context around this, via VPC subnet selection

Reply:
See https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ec2/SubnetSelection.html